lib/resty/mlcache.lua

```diff
@@ -412,15 +407,30 @@ function _M:get(key, opts, cb, ...)
         return unlock_and_ret(lock, data, nil, 2)
     end
 
+    -- opts validation
+
+    local ttl, neg_ttl = check_opts(self, opts)
+
     -- still not in shm, we are responsible for running the callback
 
-    local ok, err = pcall(cb, ...)
+    local ok, err, new_ttl = pcall(cb, ...)
     if not ok then
         return unlock_and_ret(lock, nil, "callback threw an error: " .. err)
     end
 
     data = err
 
+    -- override ttl / neg_ttl
+
+    if new_ttl and type(new_ttl) == "number" and new_ttl >= 0 then
```

Comment:
I think the following condition should suffice:
```lua
if type(new_ttl) == "number" and new_ttl >= 0 then
```

Because if `new_ttl = nil`, then `type(new_ttl) == "nil"`, which makes the first condition fail already.